### PR TITLE
feat(core): open app in electron app entry

### PR DESCRIPTION
**Pull request by pengx17**
fix PD-208
fix PD-210
fix PD-209
fix AF-1495

### CODE REVIEW

**Comment by pengx17** [comment]
* **#8637** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8637?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pengx17 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/modules/open-in-app/services/index.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    this.initialUrl = window.location.href;
+
+    const maybeDocLink = resolveLinkToDoc(this.initialUrl);
+    let shouldOpenInApp = false;
+
+    if (
+      maybeDocLink &&
+      this.workspacesService.list.workspaces$.value.find(
+        w => w.id === maybeDocLink.workspaceId
+      )?.flavour === WorkspaceFlavour.AFFINE_CLOUD
```
Comment (EYHN) [inline packages/frontend/core/src/modules/open-in-app/services/index.ts]: Maybe the user is not logged in at this time?
We just need to check if the workspace is not a local workspace.

Comment (pengx17) [reply packages/frontend/core/src/modules/open-in-app/services/index.ts]: I see, this list only makes sense if the user is logged in? I will read the value from localstorage instead.

File: packages/frontend/core/src/modules/open-in-app/views/open-in-app-page.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        <img src={appIcon} alt={appName} width={120} height={120} />
+
+        <div className={styles.prompt}>
+          <Trans i18nKey="com.affine.auth.open.affine.prompt">
+            Open {appName} app now
+          </Trans>
+        </div>
+
+        <div className={styles.promptLinks}>
+          {maybeDocLink && (
```
Comment (EYHN) [inline packages/frontend/core/src/modules/open-in-app/views/open-in-app-page.tsx]:
```suggestion
          {openHereClicked && (
```

File: packages/frontend/core/src/modules/open-in-app/services/index.ts (hunk 1 above)
Comment (EYHN) [inline packages/frontend/core/src/modules/open-in-app/services/index.ts]:
```suggestion
      !this.workspacesService.list.workspaces$.value.some(
        w => w.id === maybeDocLink.workspaceId && w.flavour === WorkspaceFlavour.LOCAL
      )
```

Comment (pengx17) [reply packages/frontend/core/src/modules/open-in-app/services/index.ts]: will use `getLocalWorkspaceIds` instead

**Review by EYHN — APPROVED** [review]

**Pull request merged**